mmv1/third_party/terraform/resources/resource_composer_environment.go.erb

```diff
@@ -332,6 +337,48 @@ func resourceComposerEnvironment() *schema.Resource {
 								},
 							},
 						},
+						"recovery_config": {
+							Type:         schema.TypeList,
+							Optional:     true,
+							AtLeastOneOf: composerConfigKeys,
+							MaxItems:     1,
+							Description:  `The recovery configuration settings for the Cloud Composer environment`,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"scheduled_snapshots_config": {
+										Type:         schema.TypeList,
+										Optional:     true,
+										AtLeastOneOf: recoveryConfigKeys,
+										Description:  `The configuration settings for scheduled snapshots.`,
+										MaxItems:     1,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"enabled": {
+													Type:        schema.TypeBool,
+													Optional:    true,
+													Description: `When enabled, Cloud Composer periodically saves snapshots of your environment to a Cloud Storage bucket.`,
+												},
+												"snapshot_location": {
+													Type:        schema.TypeString,
+													Optional:    true,
+													Description: `the URI of a bucket folder where to save the snapshot.`,
+												},
+												"snapshot_creation_schedule": {
+													Type:        schema.TypeString,
+													Optional:    true,
+													Description: `Snapshot schedule, in the unix-cron format.`,
+												},
+												"time_zone": {
+													Type:        schema.TypeString,
+													Optional:    true,
+													Description: `A time zone for the schedule. This value is a time offset and does not take into account daylight saving time changes. Valid values are from UTC-12 to UTC+12. Examples: UTC, UTC-01, UTC+03.`,
```

Comment:
Do we want to make one of the fields required or have `AtLeastOneOf` validation for them to prevent users from send empty block `scheduled_snapshots_config{}`?

Reply:
sending an empty block will be treated as a disabling scheduled snapshots: {enabled: false}